src/subscription.ts

```diff
@@ -528,11 +564,17 @@ export class Subscription extends Subscriber {
    *   const apiResponse = data[0];
    * });
    */
-  getMetadata(gaxOpts?, callback?) {
-    if (is.fn(gaxOpts)) {
-      callback = gaxOpts;
-      gaxOpts = {};
-    }
+  getMetadata(callback?: RequestCallback<Subscription>):
+      void|Promise<Subscription>;
+  getMetadata(gaxOpts: CallOptions, callback: RequestCallback<Subscription>):
+      void;
+  getMetadata(
+      gaxOptsOrCallback?: CallOptions|RequestCallback<Subscription>,
+      callback?: RequestCallback<Subscription>): void|Promise<Subscription> {
```

Comment:
I believe in all cases above `google.pubsub.v1.Subscription` should be used instead of `Subscription`